src/database/sql_database.cc

```diff
@@ -1069,7 +1100,16 @@ std::vector<std::shared_ptr<CdsObject>> SQLDatabase::search(const SearchParam& p
         limit = fmt::format(" LIMIT {} OFFSET {}", (requestedCount == 0 ? 10000000000 : requestedCount), startingIndex);
     }
 
-    auto retrievalSQL = fmt::format("SELECT DISTINCT {} {} FROM {} {} WHERE {}{}{}", sql_search_columns, addColumns, sql_search_query, addJoin, searchSQL, orderBy, limit);
+    std::string retrievalSQL;
+    if (rootContainer) {
+        // Use faster, non-recursive search for root container
+        retrievalSQL = fmt::format("SELECT DISTINCT {} {} FROM {} {} WHERE {}{}{}", sql_search_columns, addColumns, sql_search_query, addJoin, searchSQL, orderBy, limit);
+    } else {
+        // Use recursive container search
+        const std::string retrievalSelect = fmt::format("DISTINCT {} {}", sql_search_columns, addColumns);
+        retrievalSQL = fmt::format(sql_search_container_query_format, param.getContainerID(), retrievalSelect);
+        retrievalSQL += fmt::format(" {} WHERE {}{}{}", addJoin, searchSQL, orderBy, limit);
```

Comment:
You may need a second property to avoid that `+=` here - or add empty parameters above.

Reply:
Feels wasteful to create an near duplicate property just to avoid a format call and concatenation.